server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -129,31 +133,35 @@ public Pool transform(ActivationKeyPool akp) {
     @Path("{activation_key_id}")
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.APPLICATION_JSON)
-    public ActivationKeyData updateActivationKey(
+    public ActivationKeyDTO updateActivationKey(
         @PathParam("activation_key_id") @Verify(ActivationKey.class) String activationKeyId,
-        @ApiParam(name = "update", required = true) ActivationKeyData update) {
+        @ApiParam(name = "update", required = true) ActivationKeyDTO update) {
 
         ActivationKey toUpdate = activationKeyCurator.verifyAndLookupKey(activationKeyId);
         if (update.getName() != null) {
             toUpdate.setName(update.getName());
         }
+
         String serviceLevel = update.getServiceLevel();
         if (serviceLevel != null) {
             serviceLevelValidator.validate(toUpdate.getOwner(), serviceLevel);
             toUpdate.setServiceLevel(serviceLevel);
         }
+
         if (update.getReleaseVersion() != null) {
-            toUpdate.setReleaseVer(update.getReleaseVersion());
+            toUpdate.setReleaseVer(new Release(update.getReleaseVersion()));
         }
+
         if (update.getDescription() != null) {
             toUpdate.setDescription(update.getDescription());
         }
+
         if (update.isAutoAttach() != null) {
             toUpdate.setAutoAttach(update.isAutoAttach());
         }
         activationKeyCurator.merge(toUpdate);
```

Comment:
Pre-existing oversight here. Update this so we're updating the entity we're about to output. There's no guarantee that the entity in question will be merged in-place.

```
toUpdate = this.activationKeyCurator.merge(toUpdate);
```